dali/pipeline/executor/workspace_policy.h

```diff
@@ -219,28 +200,30 @@ inline void SetupStreamsAndEvents<OpType::GPU>(
   // 1 stream for GPU ops, even though we may not fill
   // the whole GPU with just I/O pipeline kernels
   // by doing so.
-  ws.set_stream(gpu_op_stream);
-  for (const auto &p : node.parents) {
-    if (graph.NodeType(p) == OpType::MIXED) {
-      const auto &parent_op = graph.Node(p);
-      // We need to block on this op's event to
-      // make sure that we respect the dependency
-      ws.AddParentEvent(mixed_op_events[parent_op.partition_index][idxs[OpType::MIXED]]);
+  if (device_id != CPU_ONLY_DEVICE_ID) {
```

Comment:
Do we need that? Will SetupStreamsAndEvents<OpType::GPU> be called at all for the CPU only pipeline?

Reply:
Hmm, probably not, as we are not supposed to run GPU ops, so there won't be any device workspaces created, but for completeness I would leave it like that.